neqo-common/src/log.rs

```diff
@@ -80,28 +80,93 @@ macro_rules! log_invoke {
         ::neqo_common::do_log!($lvl, "[{}] {}", $ctx, format!($($arg)*));
     } )
 }
+
+#[cfg(not(feature = "bench"))]
 #[macro_export]
 macro_rules! qerror {
     ([$ctx:expr], $($arg:tt)*) => (::neqo_common::log_invoke!(::log::Level::Error, $ctx, $($arg)*););
     ($($arg:tt)*) => ( { ::neqo_common::log::init(); ::neqo_common::do_log!(::log::Level::Error, $($arg)*); } );
 }
+
+#[cfg(feature = "bench")]
+#[macro_export]
+macro_rules! qerror {
+    ([$ctx:expr], $($arg:tt)*) => {
+        ()
```

Comment:
According to [this](https://stackoverflow.com/a/44722605), you could do something like this:

```rust
([$ctx:expr], $($arg:tt)*) => {
  {&($ctx, $($arg),*)}
}
```

You would have to validate that the compiler turned that into a noop.

Reply:
@martinthomson going to this construction
```rs
#[cfg(feature = "bench")]
#[macro_export]
macro_rules! qerror {
    ([$ctx:expr], $($arg:tt)*) => { &($ctx, $($arg)*); };
    ($($arg:tt)*) => { &($($arg)*); };
}
```
omits the warnings, but results in borrow issues like this:
```
error[E0382]: use of moved value: `alpn`
   --> neqo-crypto/src/agent.rs:111:8
    |
95  |     let alpn = match (pre, alpn_state) {
    |         ---- move occurs because `alpn` has type `Option<String>`, which does not implement the `Copy` trait
...
110 |     qtrace!([format!("{fd:p}")], "got ALPN {:?}", alpn);
    |                                                   ---- value moved here
111 |     Ok(alpn)
    |        ^^^^ value used here after move
    |
help: consider cloning the value if the performance cost is acceptable
    |
110 |     qtrace!([format!("{fd:p}")], "got ALPN {:?}", alpn.clone());
    |                                                       ++++++++
```